src/main/java/backend/Repo.java

```diff
@@ -108,4 +115,68 @@ public static Repo getRepo() throws IOException {
     Git git = new Git(repo);
     return new Repo(git);
   }
+
+  /**
+   * public Set < String > diffSinceTag(String tagName) gets the set of filenames that were changed
+   * since the commit tagged with tagName.
+   * Input: tagName: get diff since this name.
+   * Output: -- set of filenames that have been changed since the tagged commit List of filenames
+   * that have been changed since the tagged commit - Can be printed or converted to string.
+   */
+
+  public static List<DiffEntry> diffSinceTag(String tagName) throws IOException, GitAPIException {
```

Comment:
I think DiffEntry objects would be better for us. Has a bunch on methods related to it that help us have more details about the differences tree and path. It even has a to String method. 

Check the link `http://download.eclipse.org/jgit/docs/jgit-2.0.0.201206130900-r/apidocs/org/eclipse/jgit/diff/DiffEntry.html```
It you still think we need a Set let me know so that I can perform the changes. 
If you still need Set<String>

Reply:
Hrm yeah, DiffEntry does have a lot of nice methods, eh?

I was just thinking it would be easier to compare Set returned by the `Doc`s method and makes sure we don't have to import any jGit dependencies in the other modules.

If you think we'll be able to make use of it in the near future though, let's keep it as a DiffEntry and I'll make the conversions in `Core`.

Reply:
I thought in the near future we can use some of the methods it comes with.
I didn't want to lose the information that comes with the diffEntry objects, but it's however you want. 
Here is the code I had to perform the changes 
```
Set DiffEntries = new Hashset<String>();
        for (DiffEntry fileNames : diffs)
 {
          String foo = entry.toString();
          DiffEntries.add(foo);
}
```